### ISSUE TITLE
dmtcp_restart doesn't implement --join and --new-coordinator

### ISSUE THREAD

**Issue by gc00** [dmtcp/dmtcp#134]
Apparently, theUsage for dmtcp_restart was copied from dmtcp_launch.  I assume that we should implement --join and --new-coordinator, but we don't seem to.  Yet, dmtcp_restart_script.sh tries to use --join for the "multipleHosts" case.
In particular, `dmtcp_restart --new-coordinator` has a clear use case,  Suppose we want to restart, and we want to be sure that there was time for the old coordinator to exit first.

Finally, in `theUsage' for dmtcp_launch and dmtcp_restart, we don't make clear the default policy.  The default policy is neither --join nor --new-coordinator.  But it's:  if a coordinator exists then join it; otherwise start a new coodrinator.  The variable `allowedModes` in dmtcp_launch tries to keep track of this.

This bug should be easy to fix, and we should do so before the 2.4.0 release.

**Comment by karya0** [comment]
I think I am missing something here. I just took a look at https://github.com/dmtcp/dmtcp/blob/master/src/dmtcp_restart.cpp#L461 and both these options are supported by dmtcp_restart.

**Comment by gc00** [comment]
You're right.  My mistake.  Probably too much debugging late at night. Sorry.

In any case, I'll change the title of the issue.  It seems like we should still make clear
in `theUsage` that neither --join nor --new-coordinator is the default behavior.
